walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/ManifestIssuanceRequest.kt

```diff
@@ -59,23 +57,55 @@ internal class ManifestIssuanceRequest(
         return !validationResult.isFailure
     }
 
-    override suspend fun cancel(message: String?): Result<Unit> {
-        return try {
+    override suspend fun cancel(message: String?): VerifiedIdResult<Unit> {
+        return getResult {
             val issuanceCompletionResponse = requestState?.let {
                 IssuanceCompletionResponse(
                     IssuanceCompletionResponse.IssuanceCompletionCode.ISSUANCE_FAILED,
                     it,
                     IssuanceCompletionResponse.IssuanceCompletionErrorDetails.USER_CANCELED
                 )
             }
-            val result = VerifiedIdRequester.sendIssuanceCallback(
+            VerifiedIdRequester.sendIssuanceCallback(
                 issuanceCompletionResponse,
                 issuanceCallbackUrl
             )
-            Result.success(result)
-        } catch (exception: WalletLibraryException) {
-            Result.failure(exception)
         }
+    }
 
+    private suspend fun sendIssuanceCallbackIfRequestStateAndCallbackExists(
+        requestState: String?,
+        result: VerifiedIdResult<VerifiedId>,
+        issuanceCallbackUrl: String?
+    ) {
+        if (requestState == null || issuanceCallbackUrl == null) {
+            return
+        }
+        var issuanceCompletionCode: IssuanceCompletionResponse.IssuanceCompletionCode =
+            IssuanceCompletionResponse.IssuanceCompletionCode.ISSUANCE_FAILED
+        var issuanceCompletionErrorDetails: IssuanceCompletionResponse.IssuanceCompletionErrorDetails? =
+            null
+        result.fold(
+            onSuccess = {
+                issuanceCompletionCode =
+                    IssuanceCompletionResponse.IssuanceCompletionCode.ISSUANCE_SUCCESSFUL
+            },
+            onFailure = {
+                issuanceCompletionCode =
+                    IssuanceCompletionResponse.IssuanceCompletionCode.ISSUANCE_FAILED
+                issuanceCompletionErrorDetails = when (it) {
+                    is VerifiedIdException -> IssuanceCompletionResponse.IssuanceCompletionErrorDetails.ISSUANCE_SERVICE_ERROR
```

Comment:
Do we support other messages?

Reply:
yes, we support 4 messages, but they are handled in respective classes for contract fetching and issuance completion.